In [43]:
import pandas as pd

import numpy as np

In [44]:
description = pd.read_csv(r"D:\My Statistics\Pandas\New folder\Maven Toys Data\data_dictionary.csv")

In [45]:
stores = pd.read_csv(r"D:\My Statistics\Pandas\New folder\Maven Toys Data\stores.csv")

In [46]:
sales = pd.read_csv(r"D:\My Statistics\Pandas\New folder\Maven Toys Data\sales.csv")

In [47]:
products = pd.read_csv(r"D:\My Statistics\Pandas\New folder\Maven Toys Data\products.csv")

In [48]:
inventory = pd.read_csv(r"D:\My Statistics\Pandas\New folder\Maven Toys Data\inventory.csv")

In [49]:
date = pd.read_csv(r"D:\My Statistics\Pandas\New folder\Maven Toys Data\calendar.csv")

In [50]:
sales.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 829262 entries, 0 to 829261
Data columns (total 5 columns):
 #   Column      Non-Null Count   Dtype 
---  ------      --------------   ----- 
 0   Sale_ID     829262 non-null  int64 
 1   Date        829262 non-null  object
 2   Store_ID    829262 non-null  int64 
 3   Product_ID  829262 non-null  int64 
 4   Units       829262 non-null  int64 
dtypes: int64(4), object(1)
memory usage: 31.6+ MB


In [51]:
products.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35 entries, 0 to 34
Data columns (total 5 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Product_ID        35 non-null     int64 
 1   Product_Name      35 non-null     object
 2   Product_Category  35 non-null     object
 3   Product_Cost      35 non-null     object
 4   Product_Price     35 non-null     object
dtypes: int64(1), object(4)
memory usage: 1.5+ KB


In [53]:
sale_prod = pd.merge(sales , products , on = 'Product_ID')

In [54]:
sale_prod.head()

,Sale_ID,Date,Store_ID,Product_ID,Units,Product_Name,Product_Category,Product_Cost,Product_Price
0,1,2022-01-01,24,4,1,Chutes & Ladders,Games,$9.99,$12.99
1,2,2022-01-01,28,1,1,Action Figure,Toys,$9.99,$15.99
2,3,2022-01-01,6,8,1,Deck Of Cards,Games,$3.99,$6.99
3,4,2022-01-01,48,7,1,Dart Gun,Sports & Outdoors,$11.99,$15.99
4,5,2022-01-01,44,18,1,Lego Bricks,Toys,$34.99,$39.99


In [55]:
stores.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50 entries, 0 to 49
Data columns (total 5 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Store_ID         50 non-null     int64 
 1   Store_Name       50 non-null     object
 2   Store_City       50 non-null     object
 3   Store_Location   50 non-null     object
 4   Store_Open_Date  50 non-null     object
dtypes: int64(1), object(4)
memory usage: 2.1+ KB


In [56]:
toy_sales = pd.merge(sale_prod, stores , on = 'Store_ID')

In [57]:
inventory.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1593 entries, 0 to 1592
Data columns (total 3 columns):
 #   Column         Non-Null Count  Dtype
---  ------         --------------  -----
 0   Store_ID       1593 non-null   int64
 1   Product_ID     1593 non-null   int64
 2   Stock_On_Hand  1593 non-null   int64
dtypes: int64(3)
memory usage: 37.5 KB


In [58]:
toy_sales.head()

,Sale_ID,Date,Store_ID,Product_ID,Units,Product_Name,Product_Category,Product_Cost,Product_Price,Store_Name,Store_City,Store_Location,Store_Open_Date
0,1,2022-01-01,24,4,1,Chutes & Ladders,Games,$9.99,$12.99,Maven Toys Aguascalientes 1,Aguascalientes,Downtown,2010-07-31
1,2,2022-01-01,28,1,1,Action Figure,Toys,$9.99,$15.99,Maven Toys Puebla 2,Puebla,Downtown,2011-04-01
2,3,2022-01-01,6,8,1,Deck Of Cards,Games,$3.99,$6.99,Maven Toys Mexicali 1,Mexicali,Commercial,2003-12-13
3,4,2022-01-01,48,7,1,Dart Gun,Sports & Outdoors,$11.99,$15.99,Maven Toys Saltillo 2,Saltillo,Commercial,2016-03-23
4,5,2022-01-01,44,18,1,Lego Bricks,Toys,$34.99,$39.99,Maven Toys Puebla 3,Puebla,Residential,2014-12-27


In [59]:
toy_sales.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 829262 entries, 0 to 829261
Data columns (total 13 columns):
 #   Column            Non-Null Count   Dtype 
---  ------            --------------   ----- 
 0   Sale_ID           829262 non-null  int64 
 1   Date              829262 non-null  object
 2   Store_ID          829262 non-null  int64 
 3   Product_ID        829262 non-null  int64 
 4   Units             829262 non-null  int64 
 5   Product_Name      829262 non-null  object
 6   Product_Category  829262 non-null  object
 7   Product_Cost      829262 non-null  object
 8   Product_Price     829262 non-null  object
 9   Store_Name        829262 non-null  object
 10  Store_City        829262 non-null  object
 11  Store_Location    829262 non-null  object
 12  Store_Open_Date   829262 non-null  object
dtypes: int64(4), object(9)
memory usage: 82.2+ MB


In [60]:
toy_sales['Date'] = pd.to_datetime(toy_sales['Date'])

In [61]:
toy_sales['Date'].info()

<class 'pandas.core.series.Series'>
RangeIndex: 829262 entries, 0 to 829261
Series name: Date
Non-Null Count   Dtype         
--------------   -----         
829262 non-null  datetime64[ns]
dtypes: datetime64[ns](1)
memory usage: 6.3 MB


In [17]:
toy_sales.head()

,Sale_ID,Date,Store_ID,Product_ID,Units,Product_Name,Product_Category,Product_Cost,Product_Price,Store_Name,Store_City,Store_Location,Store_Open_Date
0,1,2022-01-01,24,4,1,Chutes & Ladders,Games,$9.99,$12.99,Maven Toys Aguascalientes 1,Aguascalientes,Downtown,2010-07-31
1,2,2022-01-01,28,1,1,Action Figure,Toys,$9.99,$15.99,Maven Toys Puebla 2,Puebla,Downtown,2011-04-01
2,3,2022-01-01,6,8,1,Deck Of Cards,Games,$3.99,$6.99,Maven Toys Mexicali 1,Mexicali,Commercial,2003-12-13
3,4,2022-01-01,48,7,1,Dart Gun,Sports & Outdoors,$11.99,$15.99,Maven Toys Saltillo 2,Saltillo,Commercial,2016-03-23
4,5,2022-01-01,44,18,1,Lego Bricks,Toys,$34.99,$39.99,Maven Toys Puebla 3,Puebla,Residential,2014-12-27


In [18]:
toy_sales.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 829262 entries, 0 to 829261
Data columns (total 13 columns):
 #   Column            Non-Null Count   Dtype         
---  ------            --------------   -----         
 0   Sale_ID           829262 non-null  int64         
 1   Date              829262 non-null  datetime64[ns]
 2   Store_ID          829262 non-null  int64         
 3   Product_ID        829262 non-null  int64         
 4   Units             829262 non-null  int64         
 5   Product_Name      829262 non-null  object        
 6   Product_Category  829262 non-null  object        
 7   Product_Cost      829262 non-null  object        
 8   Product_Price     829262 non-null  object        
 9   Store_Name        829262 non-null  object        
 10  Store_City        829262 non-null  object        
 11  Store_Location    829262 non-null  object        
 12  Store_Open_Date   829262 non-null  object        
dtypes: datetime64[ns](1), int64(4), object(8)
memory usage: 82.

In [19]:
toy_sales.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 829262 entries, 0 to 829261
Data columns (total 13 columns):
 #   Column            Non-Null Count   Dtype         
---  ------            --------------   -----         
 0   Sale_ID           829262 non-null  int64         
 1   Date              829262 non-null  datetime64[ns]
 2   Store_ID          829262 non-null  int64         
 3   Product_ID        829262 non-null  int64         
 4   Units             829262 non-null  int64         
 5   Product_Name      829262 non-null  object        
 6   Product_Category  829262 non-null  object        
 7   Product_Cost      829262 non-null  object        
 8   Product_Price     829262 non-null  object        
 9   Store_Name        829262 non-null  object        
 10  Store_City        829262 non-null  object        
 11  Store_Location    829262 non-null  object        
 12  Store_Open_Date   829262 non-null  object        
dtypes: datetime64[ns](1), int64(4), object(8)
memory usage: 82.

In [20]:
toy_sales['Product_Price'] = toy_sales['Product_Price'].replace('[\$,]', '', regex=True).astype(float)

In [21]:
toy_sales['Product_Cost'] = toy_sales['Product_Cost'].replace('[\$,]' , '' , regex = True).astype(float)

In [22]:
toy_sales.head()

,Sale_ID,Date,Store_ID,Product_ID,Units,Product_Name,Product_Category,Product_Cost,Product_Price,Store_Name,Store_City,Store_Location,Store_Open_Date
0,1,2022-01-01,24,4,1,Chutes & Ladders,Games,9.99,12.99,Maven Toys Aguascalientes 1,Aguascalientes,Downtown,2010-07-31
1,2,2022-01-01,28,1,1,Action Figure,Toys,9.99,15.99,Maven Toys Puebla 2,Puebla,Downtown,2011-04-01
2,3,2022-01-01,6,8,1,Deck Of Cards,Games,3.99,6.99,Maven Toys Mexicali 1,Mexicali,Commercial,2003-12-13
3,4,2022-01-01,48,7,1,Dart Gun,Sports & Outdoors,11.99,15.99,Maven Toys Saltillo 2,Saltillo,Commercial,2016-03-23
4,5,2022-01-01,44,18,1,Lego Bricks,Toys,34.99,39.99,Maven Toys Puebla 3,Puebla,Residential,2014-12-27


### Finding revenue 

In [23]:
toy_sales['Revenue'] = toy_sales['Units'] * toy_sales['Product_Price']

In [25]:
# Total revenue of all product

In [26]:
total_revenue = round(toy_sales['Revenue'].sum() , 3)

print(f'${total_revenue}')

$14444572.35


In [27]:
### creating profit column

In [28]:
toy_sales['Profit'] = toy_sales['Revenue'] - toy_sales['Product_Cost']

In [29]:
net_profit = toy_sales['Profit'].sum()


print(net_profit)

6171472.97


### Finding Revenue by each product category

In [30]:
revenue_product_cat = toy_sales.groupby('Product_Category')['Revenue'].sum().sort_values(ascending = False)

print(revenue_product_cat)

Product_Category
Toys                 5093241.00
Art & Crafts         2705364.26
Electronics          2246771.25
Games                2226836.27
Sports & Outdoors    2172359.57
Name: Revenue, dtype: float64


The best selling products are toys for this store.

The revenue from Toys is about $ 5093241.

This amount is the highest revenue for this store.

In [31]:
profit_product_cat = toy_sales.groupby('Product_Category')['Profit'].sum().sort_values(ascending = False)


profit_product_cat

Product_Category
Toys                 1821172.27
Electronics          1283326.50
Art & Crafts         1271718.99
Games                 952373.33
Sports & Outdoors     842881.88
Name: Profit, dtype: float64

In [32]:
product_categor_stats = toy_sales.groupby('Product_Category').agg(Revenue =('Revenue' , 'sum') , 
                                                                  Profit = ('Profit' , 'sum'),
                                                                 Unit_Sold = ('Units' , 'sum')).sort_values('Profit' , ascending = False)

In [33]:
product_categor_stats

,Revenue,Profit,Unit_Sold
Product_Category,,,
Toys,5093241.00,1821172.27,267200
Electronics,2246771.25,1283326.50,134075
Art & Crafts,2705364.26,1271718.99,325574
Games,2226836.27,952373.33,194673
Sports & Outdoors,2172359.57,842881.88,169043


In [34]:
product_categor_stats['Profit_Percent'] = (product_categor_stats['Profit']/product_categor_stats['Profit'].sum()) * 100

In [35]:
product_categor_stats

,Revenue,Profit,Unit_Sold,Profit_Percent
Product_Category,,,,
Toys,5093241.00,1821172.27,267200,29.509524
Electronics,2246771.25,1283326.50,134075,20.794493
Art & Crafts,2705364.26,1271718.99,325574,20.606409
Games,2226836.27,952373.33,194673,15.431864
Sports & Outdoors,2172359.57,842881.88,169043,13.657710


In [36]:
product_categor_stats['Profit_Margin'] = product_categor_stats['Profit'] / product_categor_stats['Revenue']

product_categor_stats

,Revenue,Profit,Unit_Sold,Profit_Percent,Profit_Margin
Product_Category,,,,,
Toys,5093241.00,1821172.27,267200,29.509524,0.357566
Electronics,2246771.25,1283326.50,134075,20.794493,0.571187
Art & Crafts,2705364.26,1271718.99,325574,20.606409,0.470073
Games,2226836.27,952373.33,194673,15.431864,0.427680
Sports & Outdoors,2172359.57,842881.88,169043,13.657710,0.388003


### Analyse on Profit

We got 29 % of profit from selling Toys. 

The profit gaining from Electronics and Art&Crafts are not quite different.

The profit from selling video games is 15%.

However, Electronics make 57% of profit from it's revenue.
This is the higest profit margin. We should make more promotion on Eelectionics.

In [37]:
product_categor_stats.to_csv(r"D:\My Statistics\Pandas\My reults and Analyst\product_categor_stats.csv")

OSError: Cannot save file into a non-existent directory: 'D:\My Statistics\Pandas\My reults and Analyst'

### Analyze on each product

In [37]:
product_summary = toy_sales.groupby(['Product_Category' , 'Product_Name']).agg(Revenue = ('Revenue' , 'sum') , 
                                                        Profit = ('Profit' , 'sum') , 
                                                        Quantity_sold = ('Units' , 'sum'))

In [38]:
product_summary.sort_values(['Product_Category' , 'Profit'] , ascending = False)

Revenue      Profit  Quantity_sold
Product_Category  Product_Name                                                
Toys              Lego Bricks            2388882.63   708312.93          59737
                  Action Figure           926748.42   442263.39          57958
                  Animal Figures          507766.11   185588.61          39089
                  Dinosaur Figures        434889.88   167711.99          29012
                  Dino Egg                420213.64   138685.45          38236
                  Plush Pony              109705.12    61806.40           5488
                  Hot Wheels 5-Pack       135727.41    52831.17          22659
                  Mr. Potatohead           85963.95    46942.15           8605
                  Teddy Bear               83343.84    17030.18           6416
Sports & Outdoors Splash Balls            541629.52   209573.11          60248
                  Dart Gun                505092.12   190918.15          31588
                  Nerf Gun                530594.57   175196.66          26543
                  Mini Ping Pong Set      335703.96   165301.74          33604
                  Mini Basketball Hoop     66148.53    42936.35           2647
                  Foam Disk Launcher       91363.80    30123.92           7620
                  Supersoaker Water Gun   101827.07    28831.95           6793
Games             Deck Of Cards           587397.66   315746.49          84034
                  Glass Marbles           412322.82   265525.89          37518
                  Rubik's Cube            912983.28   216716.31          45672
                  Jenga                   130409.46    94101.89          13054
                  Monopoly                 67666.15    25668.17           3385
                  Chutes & Ladders         49738.71    12775.71           3829
                  Uno Card Game            21652.90    11282.89           2710
                  Classic Dominoes         44665.29    10555.98           4471
Electronics       Colorbuds              1564476.32  1054290.20         104368
                  Gamer Headphones        376938.42   143948.85          17958
                  Toy Robot               305356.51    85087.45          11749
Art & Crafts      Magic Sand              968962.02   419252.95          60598
                  Barrel O' Slime         365735.37   258120.15          91663
                  Kids Makeup Kit         488415.67   185560.15          24433
                  PlayDoh Can             308352.72   179333.06         103128
                  Etch A Sketch           255406.32   132263.37          12168
                  Playfoam                 45696.42    34476.54           4158
                  PlayDoh Toolkit         115408.72    32672.08          23128
                  PlayDoh Playset         157387.02    30040.69           6298

### finding top selling product 

In [39]:
product_summary = toy_sales.groupby('Product_Name').agg(Unit_Sold = ('Units' , 'sum') , 
                                                        Revenue = ('Revenue' , 'sum') ,
                                                        Profit = ('Profit' , 'sum')).sort_values(['Unit_Sold','Profit'] , ascending = False)

In [40]:
product_summary['Profit_Margin'] = product_summary['Profit'] / product_summary['Revenue']

product_summary['Profit_Percent'] = (product_summary['Profit'] / product_summary['Profit'].sum()) * 100

In [42]:
product_summary.to_csv(r'D:\My Statistics\Pandas\Mexico toy sales result\product_summary.csv')

Colorbuds is best selling product.

We could sold 104368 Units. This show that Colorbuds is the most popular product among our customers.

We should stock them to avoid shortage.

It is also the highest profit margin. If we sell 1 unit of Colorbuds, we get 67 net profit.

We obtain 17% of profit from selling Colorbuds.This percent is the highest profit share.

In [48]:
product_summary.sort_values('Profit' , ascending = False)

,Unit_Sold,Revenue,Profit,Profit_Margin,Profit_Percent
Product_Name,,,,,
Colorbuds,104368,1564476.32,1054290.20,0.673893,17.083283
Lego Bricks,59737,2388882.63,708312.93,0.296504,11.477210
Action Figure,57958,926748.42,442263.39,0.477221,7.166253
Magic Sand,60598,968962.02,419252.95,0.432683,6.793402
Deck Of Cards,84034,587397.66,315746.49,0.537534,5.116226
Glass Marbles,37518,412322.82,265525.89,0.643976,4.302472
Barrel O' Slime,91663,365735.37,258120.15,0.705757,4.182472
Rubik's Cube,45672,912983.28,216716.31,0.237372,3.511582
Splash Balls,60248,541629.52,209573.11,0.386931,3.395836


In [49]:
top_four_market_share = product_summary['Profit_Percent'].loc['Colorbuds':'Magic Sand'].sum()

print(top_four_market_share)

36.08122178974723


Top 4 Best Profit Produts are

Colorbuds

Lego Bricks

Action Figures

Magic Sand

Their profit share is about 37%. This amount is too much for four products. 

So, we shold keep this product selling and promotion.

In [50]:
product_summary

,Unit_Sold,Revenue,Profit,Profit_Margin,Profit_Percent
Product_Name,,,,,
Colorbuds,104368,1564476.32,1054290.20,0.673893,17.083283
PlayDoh Can,103128,308352.72,179333.06,0.581584,2.905839
Barrel O' Slime,91663,365735.37,258120.15,0.705757,4.182472
Deck Of Cards,84034,587397.66,315746.49,0.537534,5.116226
Magic Sand,60598,968962.02,419252.95,0.432683,6.793402
Splash Balls,60248,541629.52,209573.11,0.386931,3.395836
Lego Bricks,59737,2388882.63,708312.93,0.296504,11.477210
Action Figure,57958,926748.42,442263.39,0.477221,7.166253
Rubik's Cube,45672,912983.28,216716.31,0.237372,3.511582


In [51]:
low_sale_qty = product_summary['Unit_Sold'].quantile(0.25)

In [52]:
product_price = toy_sales[['Product_Name' ,'Product_Price']].drop_duplicates()


In [53]:
product_price

,Product_Name,Product_Price
0,Chutes & Ladders,12.99
1,Action Figure,15.99
2,Deck Of Cards,6.99
3,Dart Gun,15.99
4,Lego Bricks,39.99
5,Splash Balls,8.99
7,Animal Figures,12.99
8,Mini Ping Pong Set,9.99
13,Dino Egg,10.99
15,Nerf Gun,19.99


In [54]:
product_summary = pd.merge(product_summary , product_price , on = 'Product_Name')

In [55]:
product_summary

,Product_Name,Unit_Sold,Revenue,Profit,Profit_Margin,Profit_Percent,Product_Price
0,Colorbuds,104368,1564476.32,1054290.20,0.673893,17.083283,14.99
1,PlayDoh Can,103128,308352.72,179333.06,0.581584,2.905839,2.99
2,Barrel O' Slime,91663,365735.37,258120.15,0.705757,4.182472,3.99
3,Deck Of Cards,84034,587397.66,315746.49,0.537534,5.116226,6.99
4,Magic Sand,60598,968962.02,419252.95,0.432683,6.793402,15.99
5,Splash Balls,60248,541629.52,209573.11,0.386931,3.395836,8.99
6,Lego Bricks,59737,2388882.63,708312.93,0.296504,11.477210,39.99
7,Action Figure,57958,926748.42,442263.39,0.477221,7.166253,15.99
8,Rubik's Cube,45672,912983.28,216716.31,0.237372,3.511582,19.99
9,Animal Figures,39089,507766.11,185588.61,0.365500,3.007201,12.99


In [56]:
high_price = product_summary['Product_Price'].quantile(0.75)

print(high_price)

19.99


In [57]:
product_summary.sort_values(['Unit_Sold' , 'Product_Price'] , ascending = [True , False])

,Product_Name,Unit_Sold,Revenue,Profit,Profit_Margin,Profit_Percent,Product_Price
34,Mini Basketball Hoop,2647,66148.53,42936.35,0.649090,0.695723,24.99
33,Uno Card Game,2710,21652.90,11282.89,0.521080,0.182823,7.99
32,Monopoly,3385,67666.15,25668.17,0.379335,0.415916,19.99
31,Chutes & Ladders,3829,49738.71,12775.71,0.256856,0.207012,12.99
30,Playfoam,4158,45696.42,34476.54,0.754469,0.558644,10.99
29,Classic Dominoes,4471,44665.29,10555.98,0.236335,0.171045,9.99
28,Plush Pony,5488,109705.12,61806.40,0.563387,1.001485,19.99
27,PlayDoh Playset,6298,157387.02,30040.69,0.190871,0.486767,24.99
26,Teddy Bear,6416,83343.84,17030.18,0.204336,0.275950,12.99
25,Supersoaker Water Gun,6793,101827.07,28831.95,0.283146,0.467181,14.99


In [58]:
low_sale_high_price = product_summary[(product_summary['Unit_Sold'] <= low_sale_qty) & (product_summary['Product_Price'] >= high_price)]

In [59]:
low_sale_high_price

,Product_Name,Unit_Sold,Revenue,Profit,Profit_Margin,Profit_Percent,Product_Price
27,PlayDoh Playset,6298,157387.02,30040.69,0.190871,0.486767,24.99
28,Plush Pony,5488,109705.12,61806.40,0.563387,1.001485,19.99
32,Monopoly,3385,67666.15,25668.17,0.379335,0.415916,19.99
34,Mini Basketball Hoop,2647,66148.53,42936.35,0.649090,0.695723,24.99


PlayDoh Playset is the worst. Its profit margin is about 19 %. 

Mini Basketball Hoop is lowest sale amount but its profit margin is the highest amont these products.

We should take some action to get better sales.

In [60]:
toy_sales.head()

,Sale_ID,Date,Store_ID,Product_ID,Units,Product_Name,Product_Category,Product_Cost,Product_Price,Store_Name,Store_City,Store_Location,Store_Open_Date,Revenue,Profit
0,1,2022-01-01,24,4,1,Chutes & Ladders,Games,9.99,12.99,Maven Toys Aguascalientes 1,Aguascalientes,Downtown,2010-07-31,12.99,3.0
1,2,2022-01-01,28,1,1,Action Figure,Toys,9.99,15.99,Maven Toys Puebla 2,Puebla,Downtown,2011-04-01,15.99,6.0
2,3,2022-01-01,6,8,1,Deck Of Cards,Games,3.99,6.99,Maven Toys Mexicali 1,Mexicali,Commercial,2003-12-13,6.99,3.0
3,4,2022-01-01,48,7,1,Dart Gun,Sports & Outdoors,11.99,15.99,Maven Toys Saltillo 2,Saltillo,Commercial,2016-03-23,15.99,4.0
4,5,2022-01-01,44,18,1,Lego Bricks,Toys,34.99,39.99,Maven Toys Puebla 3,Puebla,Residential,2014-12-27,39.99,5.0


### Creating Pivot Table

In [61]:
store_category = pd.pivot_table(toy_sales , index = 'Store_Name' , columns = 'Product_Category' ,values = 'Revenue', aggfunc = 'sum')

In [62]:
store_category['Total_Revenue'] = store_category.sum(axis = 1)

In [63]:
toy_sales['Store_Name'].nunique()

50

In [64]:
store_category.sort_values('Total_Revenue' , ascending = False)

Product_Category,Art & Crafts,Electronics,Games,Sports & Outdoors,Toys,Total_Revenue
Store_Name,,,,,,
Maven Toys Ciudad de Mexico 2,90076.29,112851.72,103915.27,81530.14,166180.01,554553.43
Maven Toys Guadalajara 3,81488.50,80031.12,67524.62,59489.39,160821.28,449354.91
Maven Toys Ciudad de Mexico 1,77326.87,42529.64,72040.97,72838.85,168819.88,433556.21
Maven Toys Toluca 1,108295.02,31583.30,52794.89,70867.01,147617.10,411157.32
Maven Toys Monterrey 2,64691.13,67519.80,61643.28,45251.07,133893.54,372998.82
Maven Toys Guadalajara 4,41273.42,77909.04,43952.90,37165.08,148166.20,348466.64
Maven Toys Hermosillo 3,69883.14,84740.47,49067.08,27958.86,113197.09,344846.64
Maven Toys Xalapa 2,82595.12,46296.85,45230.89,49633.21,120550.97,344307.04
Maven Toys Ciudad de Mexico 3,67190.32,30499.87,47150.87,72224.88,120358.72,337424.66


In [65]:
low_revenue_sale= store_category['Total_Revenue'].quantile(0.25)


print(low_sale_store)

NameError: name 'low_sale_store' is not defined

In [66]:
low_sale_stores = store_category[store_category['Total_Revenue'] <= low_revenue_sale].sort_values('Total_Revenue' , ascending = False)

In [67]:
len(low_sale_stores)

13

In [68]:
low_sale_stores

Product_Category,Art & Crafts,Electronics,Games,Sports & Outdoors,Toys,Total_Revenue
Store_Name,,,,,,
Maven Toys Chilpancingo 1,42729.77,42525.42,32224.10,27873.68,97186.76,242539.73
Maven Toys Oaxaca 1,52574.74,24710.67,36408.09,27760.51,100700.35,242154.36
Maven Toys Aguascalientes 1,32623.95,43525.87,44464.80,33311.20,86071.53,239997.35
Maven Toys Pachuca 1,60647.79,24851.90,39257.10,45394.04,67525.32,237676.15
Maven Toys Hermosillo 1,39595.74,59007.64,37842.84,30557.09,68111.87,235115.18
Maven Toys Merida 1,50810.72,36882.51,41706.60,32851.18,69846.71,232097.72
Maven Toys Zacatecas 1,59771.06,22380.88,29087.49,31727.33,87016.28,229983.04
Maven Toys Tuxtla Gutierrez 1,46897.88,30791.92,37370.12,49638.70,64999.65,229698.27
Maven Toys Toluca 2,51278.54,23614.56,30733.25,33727.38,83010.63,222364.36


Maven Toys Campeche 2 is the lowest sale revenue. 

We need to think about somethings such as store situations , employee services , store manager , population from Campeche.

The average salary of people around Campeche 2 store. 


In [69]:
Higher_sale_amount = store_category['Total_Revenue'].quantile(0.75)

print(Higher_sale_amount)

321049.41500000004


In [70]:
Mid_sale_amount = store_category['Total_Revenue'].quantile(0.5)

print(Mid_sale_amount)

270882.395


In [71]:
low_sale_amount = store_category['Total_Revenue'].quantile(0.25)

print(low_sale_amount)

243906.8725


In [72]:
Higher_sales_stores = store_category[store_category['Total_Revenue'] > Higher_sale_amount].sort_values('Total_Revenue' , ascending = False)   # greater than 75% quantile

In [73]:
Mid_sales_stores = store_category[(store_category['Total_Revenue'] > Mid_sale_amount) & 
    (store_category['Total_Revenue'] <= Higher_sale_amount)].sort_values('Total_Revenue' , ascending = False) # between 50 and 75

In [74]:
low_sales_stores =store_category[(store_category['Total_Revenue'] > low_sale_amount) & 
    (store_category['Total_Revenue'] <= Mid_sale_amount)].sort_values('Total_Revenue' , ascending = False) # between 25 and 50

In [75]:
lowes_sales_stores = store_category[store_category['Total_Revenue'] <= low_sale_amount].sort_values('Total_Revenue' , ascending = False) # less and equal 25%

In [76]:
print('The number of High Sales Stores')
len(Higher_sales_stores)

The number of High Sales Stores


13

In [77]:
print('The numwer of Mid Sales Stores')
len(Mid_sales_stores)

The numwer of Mid Sales Stores


12

In [78]:
print('The number of Lowest Sales Stores')
len(lowes_sales_stores)

The number of Lowest Sales Stores


13

In [79]:
print('The number of Low Sales Stores')
len(low_sales_stores)

The number of Low Sales Stores


12

In [80]:
Higher_sales_stores

Product_Category,Art & Crafts,Electronics,Games,Sports & Outdoors,Toys,Total_Revenue
Store_Name,,,,,,
Maven Toys Ciudad de Mexico 2,90076.29,112851.72,103915.27,81530.14,166180.01,554553.43
Maven Toys Guadalajara 3,81488.50,80031.12,67524.62,59489.39,160821.28,449354.91
Maven Toys Ciudad de Mexico 1,77326.87,42529.64,72040.97,72838.85,168819.88,433556.21
Maven Toys Toluca 1,108295.02,31583.30,52794.89,70867.01,147617.10,411157.32
Maven Toys Monterrey 2,64691.13,67519.80,61643.28,45251.07,133893.54,372998.82
Maven Toys Guadalajara 4,41273.42,77909.04,43952.90,37165.08,148166.20,348466.64
Maven Toys Hermosillo 3,69883.14,84740.47,49067.08,27958.86,113197.09,344846.64
Maven Toys Xalapa 2,82595.12,46296.85,45230.89,49633.21,120550.97,344307.04
Maven Toys Ciudad de Mexico 3,67190.32,30499.87,47150.87,72224.88,120358.72,337424.66


In [81]:
Mid_sales_stores

Product_Category,Art & Crafts,Electronics,Games,Sports & Outdoors,Toys,Total_Revenue
Store_Name,,,,,,
Maven Toys Guanajuato 1,41975.12,47477.23,49190.55,42977.89,132295.81,313916.60
Maven Toys Campeche 1,52691.79,26754.51,62632.04,80893.73,88814.37,311786.44
Maven Toys Ciudad Victoria 1,55622.93,65608.32,36589.87,21231.09,115751.78,294803.99
Maven Toys Mexicali 1,48109.67,104047.75,45010.63,35691.65,61159.72,294019.42
Maven Toys Mexicali 2,41508.07,42860.73,41833.46,38894.37,127059.80,292156.43
Maven Toys Monterrey 3,37798.88,43613.69,68564.21,47083.79,88753.67,285814.24
Maven Toys Puebla 2,66113.24,29422.49,33115.94,42975.84,110989.36,282616.87
Maven Toys Guanajuato 2,46138.23,51110.15,41579.81,40379.21,99719.27,278926.67
Maven Toys Monterrey 1,52697.81,43667.83,32749.08,34403.79,114440.63,277959.14


In [82]:
low_sales_stores

Product_Category,Art & Crafts,Electronics,Games,Sports & Outdoors,Toys,Total_Revenue
Store_Name,,,,,,
Maven Toys Chihuahua 2,53671.16,23108.50,47366.73,71573.12,72985.23,268704.74
Maven Toys Puebla 3,34425.84,46807.09,35632.66,27792.95,122901.01,267559.55
Maven Toys Xalapa 1,37788.82,45681.51,37350.55,45413.00,99578.85,265812.73
Maven Toys San Luis Potosi 1,60622.70,38681.27,34279.65,37961.48,91844.55,263389.65
Maven Toys Guadalajara 2,52375.39,49580.35,40552.35,32838.45,87088.48,262435.02
Maven Toys Guadalajara 1,60242.69,22958.99,47414.16,40246.17,90980.88,261842.89
Maven Toys Chetumal 1,41758.12,36392.38,36827.68,42519.64,101421.53,258919.35
Maven Toys Puebla 1,48596.97,54277.30,47864.80,31923.63,75871.17,258533.87
Maven Toys Villahermosa 1,55024.69,39722.27,40246.14,39793.39,83424.17,258210.66


In [83]:
lowest_sales_stores

Product_Category,Art & Crafts,Electronics,Games,Sports & Outdoors,Toys,Total_Revenue
Store_Name,,,,,,
Maven Toys Chilpancingo 1,42729.77,42525.42,32224.10,27873.68,97186.76,242539.73
Maven Toys Oaxaca 1,52574.74,24710.67,36408.09,27760.51,100700.35,242154.36
Maven Toys Aguascalientes 1,32623.95,43525.87,44464.80,33311.20,86071.53,239997.35
Maven Toys Pachuca 1,60647.79,24851.90,39257.10,45394.04,67525.32,237676.15
Maven Toys Hermosillo 1,39595.74,59007.64,37842.84,30557.09,68111.87,235115.18
Maven Toys Merida 1,50810.72,36882.51,41706.60,32851.18,69846.71,232097.72
Maven Toys Zacatecas 1,59771.06,22380.88,29087.49,31727.33,87016.28,229983.04
Maven Toys Tuxtla Gutierrez 1,46897.88,30791.92,37370.12,49638.70,64999.65,229698.27
Maven Toys Toluca 2,51278.54,23614.56,30733.25,33727.38,83010.63,222364.36


### The Final Analysis on Stores and Actions To Take

We have 13 High Sales Stores , 12 Mid Sales Stores , 12 Low Sales Stores and 12 Lowest Sales Stores.

We must congraculation to our High Sales Stores and Mid Sales Stores as their sales revenue.

We need to take some actions to our Low and Lowest Sales Stores.

We need to think and take some actions to our lowest Sales Stores quickly.

We should check the situation of Maven Toys Campeche 2. We need to talk the manager and employees of Maven Toys Campeche 2.

In [84]:
toy_sales.head()

,Sale_ID,Date,Store_ID,Product_ID,Units,Product_Name,Product_Category,Product_Cost,Product_Price,Store_Name,Store_City,Store_Location,Store_Open_Date,Revenue,Profit
0,1,2022-01-01,24,4,1,Chutes & Ladders,Games,9.99,12.99,Maven Toys Aguascalientes 1,Aguascalientes,Downtown,2010-07-31,12.99,3.0
1,2,2022-01-01,28,1,1,Action Figure,Toys,9.99,15.99,Maven Toys Puebla 2,Puebla,Downtown,2011-04-01,15.99,6.0
2,3,2022-01-01,6,8,1,Deck Of Cards,Games,3.99,6.99,Maven Toys Mexicali 1,Mexicali,Commercial,2003-12-13,6.99,3.0
3,4,2022-01-01,48,7,1,Dart Gun,Sports & Outdoors,11.99,15.99,Maven Toys Saltillo 2,Saltillo,Commercial,2016-03-23,15.99,4.0
4,5,2022-01-01,44,18,1,Lego Bricks,Toys,34.99,39.99,Maven Toys Puebla 3,Puebla,Residential,2014-12-27,39.99,5.0


### Profit By location

In [85]:
profit_city = toy_sales.groupby(['Store_City','Store_Name'])['Profit'].sum().sort_index()

In [86]:
profit_city

Store_City        Store_Name                   
Aguascalientes    Maven Toys Aguascalientes 1       96268.23
Campeche          Maven Toys Campeche 1            146063.39
                  Maven Toys Campeche 2             83053.28
Chetumal          Maven Toys Chetumal 1            109427.79
Chihuahua         Maven Toys Chihuahua 1           102120.28
                  Maven Toys Chihuahua 2           118152.54
Chilpancingo      Maven Toys Chilpancingo 1         98469.65
Ciudad Victoria   Maven Toys Ciudad Victoria 1     124510.33
Cuernavaca        Maven Toys Cuernavaca 1           88167.69
Cuidad de Mexico  Maven Toys Ciudad de Mexico 1    183155.03
                  Maven Toys Ciudad de Mexico 2    277790.67
                  Maven Toys Ciudad de Mexico 3    145785.17
                  Maven Toys Ciudad de Mexico 4    138892.39
Culiacan          Maven Toys Culiacan 1            108762.85
Durango           Maven Toys Durango 1              87250.88
Guadalajara       Maven Toys Guadalaj

In [87]:
summary_city = toy_sales.groupby(['Store_City' , 'Store_Name']).agg(Profit = ('Profit' , 'sum'),
                                                                    Quantity = ('Units' , 'sum'),
                                                                    Revenue = ('Revenue' , 'sum'))

In [88]:
summary_city['Cumulative_Profit'] = summary_city['Profit'].cumsum()

In [89]:
summary_city

Profit  Quantity  \
Store_City       Store_Name                                           
Aguascalientes   Maven Toys Aguascalientes 1     96268.23     17865   
Campeche         Maven Toys Campeche 1          146063.39     24556   
                 Maven Toys Campeche 2           83053.28     16277   
Chetumal         Maven Toys Chetumal 1          109427.79     18965   
Chihuahua        Maven Toys Chihuahua 1         102120.28     18070   
                 Maven Toys Chihuahua 2         118152.54     22326   
Chilpancingo     Maven Toys Chilpancingo 1       98469.65     18327   
Ciudad Victoria  Maven Toys Ciudad Victoria 1   124510.33     20501   
Cuernavaca       Maven Toys Cuernavaca 1         88167.69     17074   
Cuidad de Mexico Maven Toys Ciudad de Mexico 1  183155.03     33479   
                 Maven Toys Ciudad de Mexico 2  277790.67     42757   
                 Maven Toys Ciudad de Mexico 3  145785.17     26634   
                 Maven Toys Ciudad de Mexico 4  138892.39     22729   
Culiacan         Maven Toys Culiacan 1          108762.85     19909   
Durango          Maven Toys Durango 1            87250.88     17422   
Guadalajara      Maven Toys Guadalajara 1       104418.15     20011   
                 Maven Toys Guadalajara 2       110509.33     20698   
                 Maven Toys Guadalajara 3       196445.75     31609   
                 Maven Toys Guadalajara 4       149572.03     24136   
Guanajuato       Maven Toys Guanajuato 1        136117.17     24740   
                 Maven Toys Guanajuato 2        115558.61     21133   
                 Maven Toys Guanajuato 3        114242.25     18444   
Hermosillo       Maven Toys Hermosillo 1        107587.82     20082   
                 Maven Toys Hermosillo 2        137616.20     23498   
                 Maven Toys Hermosillo 3        165627.17     23236   
La Paz           Maven Toys La Paz 1             80008.00     16217   
Merida           Maven Toys Merida 1             88829.47     18328   
Mexicali         Maven Toys Mexicali 1          143210.06     22858   
                 Maven Toys Mexicali 2          113718.34     21557   
Monterrey        Maven Toys Monterrey 1         119018.12     20886   
                 Maven Toys Monterrey 2         167750.82     28318   
                 Maven Toys Monterrey 3         134771.73     22076   
                 Maven Toys Monterrey 4         143751.26     22450   
Morelia          Maven Toys Morelia 1           131795.61     20895   
Oaxaca           Maven Toys Oaxaca 1             94848.93     18564   
Pachuca          Maven Toys Pachuca 1            93141.84     19185   
Puebla           Maven Toys Puebla 1            107928.63     19413   
                 Maven Toys Puebla 2            112848.51     22413   
                 Maven Toys Puebla 3            117926.23     19345   
Saltillo         Maven Toys Saltillo 1          135384.14     24010   
                 Maven Toys Saltillo 2          109528.73     18893   
San Luis Potosi  Maven Toys San Luis Potosi 1   102431.64     19635   
Santiago         Maven Toys Santiago 1          109260.25     20286   
Toluca           Maven Toys Toluca 1            177015.65     32668   
                 Maven Toys Toluca 2             85887.12     15964   
Tuxtla Gutierrez Maven Toys Tuxtla Gutierrez 1  102087.45     19773   
Villahermosa     Maven Toys Villahermosa 1      109558.90     20934   
Xalapa           Maven Toys Xalapa 1            111929.71     18927   
                 Maven Toys Xalapa 2            144087.13     25296   
Zacatecas        Maven Toys Zacatecas 1          89190.05     17196   

                                                  Revenue  Cumulative_Profit  
Store_City       Store_Name                                                   
Aguascalientes   Maven Toys Aguascalientes 1    239997.35           96268.23  
Campeche         Maven Toys Campeche 1          311786.44          242331.62  
                 Maven Toys Campeche 2    

In [90]:
summary_city = summary_city.drop('Cumulative_Profit' , axis = 1)

In [91]:
summary_city

Profit  Quantity    Revenue
Store_City       Store_Name                                                   
Aguascalientes   Maven Toys Aguascalientes 1     96268.23     17865  239997.35
Campeche         Maven Toys Campeche 1          146063.39     24556  311786.44
                 Maven Toys Campeche 2           83053.28     16277  206055.23
Chetumal         Maven Toys Chetumal 1          109427.79     18965  258919.35
Chihuahua        Maven Toys Chihuahua 1         102120.28     18070  248008.30
                 Maven Toys Chihuahua 2         118152.54     22326  268704.74
Chilpancingo     Maven Toys Chilpancingo 1       98469.65     18327  242539.73
Ciudad Victoria  Maven Toys Ciudad Victoria 1   124510.33     20501  294803.99
Cuernavaca       Maven Toys Cuernavaca 1         88167.69     17074  221587.26
Cuidad de Mexico Maven Toys Ciudad de Mexico 1  183155.03     33479  433556.21
                 Maven Toys Ciudad de Mexico 2  277790.67     42757  554553.43
                 Maven Toys Ciudad de Mexico 3  145785.17     26634  337424.66
                 Maven Toys Ciudad de Mexico 4  138892.39     22729  323957.71
Culiacan         Maven Toys Culiacan 1          108762.85     19909  251160.91
Durango          Maven Toys Durango 1            87250.88     17422  222318.78
Guadalajara      Maven Toys Guadalajara 1       104418.15     20011  261842.89
                 Maven Toys Guadalajara 2       110509.33     20698  262435.02
                 Maven Toys Guadalajara 3       196445.75     31609  449354.91
                 Maven Toys Guadalajara 4       149572.03     24136  348466.64
Guanajuato       Maven Toys Guanajuato 1        136117.17     24740  313916.60
                 Maven Toys Guanajuato 2        115558.61     21133  278926.67
                 Maven Toys Guanajuato 3        114242.25     18444  276212.56
Hermosillo       Maven Toys Hermosillo 1        107587.82     20082  235115.18
                 Maven Toys Hermosillo 2        137616.20     23498  323427.02
                 Maven Toys Hermosillo 3        165627.17     23236  344846.64
La Paz           Maven Toys La Paz 1             80008.00     16217  210897.83
Merida           Maven Toys Merida 1             88829.47     18328  232097.72
Mexicali         Maven Toys Mexicali 1          143210.06     22858  294019.42
                 Maven Toys Mexicali 2          113718.34     21557  292156.43
Monterrey        Maven Toys Monterrey 1         119018.12     20886  277959.14
                 Maven Toys Monterrey 2         167750.82     28318  372998.82
                 Maven Toys Monterrey 3         134771.73     22076  285814.24
                 Maven Toys Monterrey 4         143751.26     22450  325073.50
Morelia          Maven Toys Morelia 1           131795.61     20895  273060.05
Oaxaca           Maven Toys Oaxaca 1             94848.93     18564  242154.36
Pachuca          Maven Toys Pachuca 1            93141.84     19185  237676.15
Puebla           Maven Toys Puebla 1            107928.63     19413  258533.87
                 Maven Toys Puebla 2            112848.51     22413  282616.87
                 Maven Toys Puebla 3            117926.23     19345  267559.55
Saltillo         Maven Toys Saltillo 1          135384.14     24010  330408.90
                 Maven Toys Saltillo 2          109528.73     18893  249106.07
San Luis Potosi  Maven Toys San Luis Potosi 1   102431.64     19635  263389.65
Santiago         Maven Toys Santiago 1          109260.25     20286  277598.14
Toluca           Maven Toys Toluca 1            177015.65     32668  411157.32
                 Maven Toys Toluca 2             85887.12     15964  222364.36
Tuxtla Gutierrez Maven Toys Tuxtla Gutierrez 1  102087.45     19773  229698.27
Villahermosa     Maven Toys Villahermosa 1      109558.90     20934  258210.66
Xalapa           Maven Toys Xalapa 1            111929.71     18927  265812.73
                 Maven Toys Xalapa 2            144087.13     25296  344307.04
Zacatecas        Maven

In [92]:
summary_city['Cumulative_Profit'] = summary_city.groupby('Store_City')['Profit'].cumsum()

In [93]:
summary_city

Profit  Quantity  \
Store_City       Store_Name                                           
Aguascalientes   Maven Toys Aguascalientes 1     96268.23     17865   
Campeche         Maven Toys Campeche 1          146063.39     24556   
                 Maven Toys Campeche 2           83053.28     16277   
Chetumal         Maven Toys Chetumal 1          109427.79     18965   
Chihuahua        Maven Toys Chihuahua 1         102120.28     18070   
                 Maven Toys Chihuahua 2         118152.54     22326   
Chilpancingo     Maven Toys Chilpancingo 1       98469.65     18327   
Ciudad Victoria  Maven Toys Ciudad Victoria 1   124510.33     20501   
Cuernavaca       Maven Toys Cuernavaca 1         88167.69     17074   
Cuidad de Mexico Maven Toys Ciudad de Mexico 1  183155.03     33479   
                 Maven Toys Ciudad de Mexico 2  277790.67     42757   
                 Maven Toys Ciudad de Mexico 3  145785.17     26634   
                 Maven Toys Ciudad de Mexico 4  138892.39     22729   
Culiacan         Maven Toys Culiacan 1          108762.85     19909   
Durango          Maven Toys Durango 1            87250.88     17422   
Guadalajara      Maven Toys Guadalajara 1       104418.15     20011   
                 Maven Toys Guadalajara 2       110509.33     20698   
                 Maven Toys Guadalajara 3       196445.75     31609   
                 Maven Toys Guadalajara 4       149572.03     24136   
Guanajuato       Maven Toys Guanajuato 1        136117.17     24740   
                 Maven Toys Guanajuato 2        115558.61     21133   
                 Maven Toys Guanajuato 3        114242.25     18444   
Hermosillo       Maven Toys Hermosillo 1        107587.82     20082   
                 Maven Toys Hermosillo 2        137616.20     23498   
                 Maven Toys Hermosillo 3        165627.17     23236   
La Paz           Maven Toys La Paz 1             80008.00     16217   
Merida           Maven Toys Merida 1             88829.47     18328   
Mexicali         Maven Toys Mexicali 1          143210.06     22858   
                 Maven Toys Mexicali 2          113718.34     21557   
Monterrey        Maven Toys Monterrey 1         119018.12     20886   
                 Maven Toys Monterrey 2         167750.82     28318   
                 Maven Toys Monterrey 3         134771.73     22076   
                 Maven Toys Monterrey 4         143751.26     22450   
Morelia          Maven Toys Morelia 1           131795.61     20895   
Oaxaca           Maven Toys Oaxaca 1             94848.93     18564   
Pachuca          Maven Toys Pachuca 1            93141.84     19185   
Puebla           Maven Toys Puebla 1            107928.63     19413   
                 Maven Toys Puebla 2            112848.51     22413   
                 Maven Toys Puebla 3            117926.23     19345   
Saltillo         Maven Toys Saltillo 1          135384.14     24010   
                 Maven Toys Saltillo 2          109528.73     18893   
San Luis Potosi  Maven Toys San Luis Potosi 1   102431.64     19635   
Santiago         Maven Toys Santiago 1          109260.25     20286   
Toluca           Maven Toys Toluca 1            177015.65     32668   
                 Maven Toys Toluca 2             85887.12     15964   
Tuxtla Gutierrez Maven Toys Tuxtla Gutierrez 1  102087.45     19773   
Villahermosa     Maven Toys Villahermosa 1      109558.90     20934   
Xalapa           Maven Toys Xalapa 1            111929.71     18927   
                 Maven Toys Xalapa 2            144087.13     25296   
Zacatecas        Maven Toys Zacatecas 1          89190.05     17196   

                                                  Revenue  Cumulative_Profit  
Store_City       Store_Name                                                   
Aguascalientes   Maven Toys Aguascalientes 1    239997.35           96268.23  
Campeche         Maven Toys Campeche 1          311786.44          146063.39  
                 Maven Toys Campeche 2    

### City Analysis

In [94]:
profit_city = toy_sales.groupby('Store_City').agg(Sold_Units = ('Units' ,'sum') , 
                                                  Total_Revenue = ('Revenue' , 'sum'),
                                                  Total_Profit = ('Profit' , 'sum')).sort_values('Total_Profit' , ascending = False)

In [95]:
profit_city['Profit_Margin'] = profit_city['Total_Profit']/ profit_city['Total_Revenue']

In [96]:
profit_city

,Sold_Units,Total_Revenue,Total_Profit,Profit_Margin
Store_City,,,,
Cuidad de Mexico,125599,1649492.01,745623.26,0.452032
Monterrey,93730,1261845.70,565291.93,0.447988
Guadalajara,96454,1322099.46,560945.26,0.424284
Hermosillo,66816,903388.84,410831.19,0.454767
Guanajuato,64317,869055.83,365918.03,0.421052
Puebla,61171,808710.29,338703.37,0.418819
Toluca,48632,633521.68,262902.77,0.414986
Mexicali,44415,586175.85,256928.40,0.438313
Xalapa,44223,610119.77,256016.84,0.419617


Cuidad de Mexico is the highest sales city and the highest profit city for our store.

We should analze the population and average income of people in each city.

we must take decison based on upon results. 

The lowest sales city , La Paz should be promoted the name of our stores to be more popular in that city.

### Time Series Analysis

Sales over month

In [97]:
toy_sales.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 829262 entries, 0 to 829261
Data columns (total 15 columns):
 #   Column            Non-Null Count   Dtype         
---  ------            --------------   -----         
 0   Sale_ID           829262 non-null  int64         
 1   Date              829262 non-null  datetime64[ns]
 2   Store_ID          829262 non-null  int64         
 3   Product_ID        829262 non-null  int64         
 4   Units             829262 non-null  int64         
 5   Product_Name      829262 non-null  object        
 6   Product_Category  829262 non-null  object        
 7   Product_Cost      829262 non-null  float64       
 8   Product_Price     829262 non-null  float64       
 9   Store_Name        829262 non-null  object        
 10  Store_City        829262 non-null  object        
 11  Store_Location    829262 non-null  object        
 12  Store_Open_Date   829262 non-null  object        
 13  Revenue           829262 non-null  float64       
 14  Prof

In [98]:
toy_sales.head()

,Sale_ID,Date,Store_ID,Product_ID,Units,Product_Name,Product_Category,Product_Cost,Product_Price,Store_Name,Store_City,Store_Location,Store_Open_Date,Revenue,Profit
0,1,2022-01-01,24,4,1,Chutes & Ladders,Games,9.99,12.99,Maven Toys Aguascalientes 1,Aguascalientes,Downtown,2010-07-31,12.99,3.0
1,2,2022-01-01,28,1,1,Action Figure,Toys,9.99,15.99,Maven Toys Puebla 2,Puebla,Downtown,2011-04-01,15.99,6.0
2,3,2022-01-01,6,8,1,Deck Of Cards,Games,3.99,6.99,Maven Toys Mexicali 1,Mexicali,Commercial,2003-12-13,6.99,3.0
3,4,2022-01-01,48,7,1,Dart Gun,Sports & Outdoors,11.99,15.99,Maven Toys Saltillo 2,Saltillo,Commercial,2016-03-23,15.99,4.0
4,5,2022-01-01,44,18,1,Lego Bricks,Toys,34.99,39.99,Maven Toys Puebla 3,Puebla,Residential,2014-12-27,39.99,5.0


In [99]:
toy_sales['Month'] = toy_sales['Date'].dt.to_period('M')

In [100]:
toy_sales['Year'] = toy_sales['Date'].dt.to_period('Y')

In [101]:
toy_sales.head()

,Sale_ID,Date,Store_ID,Product_ID,Units,Product_Name,Product_Category,Product_Cost,Product_Price,Store_Name,Store_City,Store_Location,Store_Open_Date,Revenue,Profit,Month,Year
0,1,2022-01-01,24,4,1,Chutes & Ladders,Games,9.99,12.99,Maven Toys Aguascalientes 1,Aguascalientes,Downtown,2010-07-31,12.99,3.0,2022-01,2022
1,2,2022-01-01,28,1,1,Action Figure,Toys,9.99,15.99,Maven Toys Puebla 2,Puebla,Downtown,2011-04-01,15.99,6.0,2022-01,2022
2,3,2022-01-01,6,8,1,Deck Of Cards,Games,3.99,6.99,Maven Toys Mexicali 1,Mexicali,Commercial,2003-12-13,6.99,3.0,2022-01,2022
3,4,2022-01-01,48,7,1,Dart Gun,Sports & Outdoors,11.99,15.99,Maven Toys Saltillo 2,Saltillo,Commercial,2016-03-23,15.99,4.0,2022-01,2022
4,5,2022-01-01,44,18,1,Lego Bricks,Toys,34.99,39.99,Maven Toys Puebla 3,Puebla,Residential,2014-12-27,39.99,5.0,2022-01,2022


In [102]:
toy_sales['Month'].unique()

<PeriodArray>
['2022-01', '2022-02', '2022-03', '2022-04', '2022-05', '2022-06', '2022-07',
 '2022-08', '2022-09', '2022-10', '2022-11', '2022-12', '2023-01', '2023-02',
 '2023-03', '2023-04', '2023-05', '2023-06', '2023-07', '2023-08', '2023-09']
Length: 21, dtype: period[M]

In [103]:
monthly_sales = toy_sales.groupby('Month').agg(Sold_Units = ('Units' , 'sum') , 
                                               Revenue = ('Revenue' , 'sum') , 
                                               Profit = ('Profit' ,'sum')).sort_values(['Month' , 'Profit']  ,ascending =[True , False])

In [104]:
monthly_sales['Year'] = monthly_sales.index.year

In [105]:
monthly_sales.columns

Index(['Sold_Units', 'Revenue', 'Profit', 'Year'], dtype='object')

In [106]:
monthly_sales.index

PeriodIndex(['2022-01', '2022-02', '2022-03', '2022-04', '2022-05', '2022-06',
             '2022-07', '2022-08', '2022-09', '2022-10', '2022-11', '2022-12',
             '2023-01', '2023-02', '2023-03', '2023-04', '2023-05', '2023-06',
             '2023-07', '2023-08', '2023-09'],
            dtype='period[M]', name='Month')

In [107]:
monthly_sales = monthly_sales.sort_values(['Year' , 'Profit'] , ascending = (True , False))

monthly_sales

,Sold_Units,Revenue,Profit,Year
Month,,,,
2022-12,66031,877203.69,384954.49,2022
2022-04,47102,681072.98,310169.14,2022
2022-06,47178,661980.22,301485.37,2022
2022-05,46910,672369.90,286248.54,2022
2022-11,51185,661304.15,275914.74,2022
2022-10,47861,623874.39,264391.90,2022
2022-07,46177,556034.23,258554.38,2022
2022-03,39981,589485.19,257778.95,2022
2022-01,38009,542554.91,241801.38,2022


In 2022, December is the most highest sale month and March is the most highest sale in 2023.

Those two highest values are close less different.

Every year , August is the lowest sales month and April is the second highest sales.

In [108]:
monthly_sales.to_csv(r'D:\My Statistics\Pandas\My reults and Analyst\monthly_sales.csv')

OSError: Cannot save file into a non-existent directory: 'D:\My Statistics\Pandas\My reults and Analyst'

### Weekly Analysis

In [109]:
toy_sales['Week'] = toy_sales['Date'].dt.to_period('W')

In [110]:
toy_sales['Date'].dt.('D')

SyntaxError: invalid syntax (1058253744.py, line 1)

In [111]:
toy_sales['Week'] = toy_sales['Date'].dt.isocalendar().week

In [112]:
toy_sales['Day'] = toy_sales['Date'].dt.weekday

In [113]:
toy_sales['Day_Name'] = toy_sales['Date'].dt.day_name()

In [114]:
toy_sales.head()

,Sale_ID,Date,Store_ID,Product_ID,Units,Product_Name,Product_Category,Product_Cost,Product_Price,Store_Name,Store_City,Store_Location,Store_Open_Date,Revenue,Profit,Month,Year,Week,Day,Day_Name
0,1,2022-01-01,24,4,1,Chutes & Ladders,Games,9.99,12.99,Maven Toys Aguascalientes 1,Aguascalientes,Downtown,2010-07-31,12.99,3.0,2022-01,2022,52,5,Saturday
1,2,2022-01-01,28,1,1,Action Figure,Toys,9.99,15.99,Maven Toys Puebla 2,Puebla,Downtown,2011-04-01,15.99,6.0,2022-01,2022,52,5,Saturday
2,3,2022-01-01,6,8,1,Deck Of Cards,Games,3.99,6.99,Maven Toys Mexicali 1,Mexicali,Commercial,2003-12-13,6.99,3.0,2022-01,2022,52,5,Saturday
3,4,2022-01-01,48,7,1,Dart Gun,Sports & Outdoors,11.99,15.99,Maven Toys Saltillo 2,Saltillo,Commercial,2016-03-23,15.99,4.0,2022-01,2022,52,5,Saturday
4,5,2022-01-01,44,18,1,Lego Bricks,Toys,34.99,39.99,Maven Toys Puebla 3,Puebla,Residential,2014-12-27,39.99,5.0,2022-01,2022,52,5,Saturday


In [115]:
weekly_trend = toy_sales.groupby(['Week' ,'Day_Name']).agg(Revenue = ('Revenue' ,'sum') ,
                                                           Profit = ('Profit' , 'sum') , 
                                                           Unit_Solds = ('Units' , 'sum')).sort_values(['Week' , 'Profit'] , ascending = (True, False))

In [116]:
weekly_trend

Revenue    Profit  Unit_Solds
Week Day_Name                                 
1    Friday     72196.33  30764.42        4767
     Thursday   57655.63  27647.13        3937
     Saturday   50131.21  21208.68        3979
     Wednesday  38822.34  16304.54        2766
     Tuesday    34352.02  15214.77        2498
...                  ...       ...         ...
52   Friday     30857.12  13556.55        2288
     Thursday   28013.86  11600.10        2114
     Tuesday    20781.01  10229.66        1599
     Monday     21157.40   8888.92        1660
     Wednesday  19095.35   8200.10        1465

[364 rows x 3 columns]

In [117]:
toy_sales['Year_Week'] = toy_sales['Date'].dt.strftime('%G-W%V')

In [118]:
toy_sales.head()

,Sale_ID,Date,Store_ID,Product_ID,Units,Product_Name,Product_Category,Product_Cost,Product_Price,Store_Name,...,Store_Location,Store_Open_Date,Revenue,Profit,Month,Year,Week,Day,Day_Name,Year_Week
0,1,2022-01-01,24,4,1,Chutes & Ladders,Games,9.99,12.99,Maven Toys Aguascalientes 1,...,Downtown,2010-07-31,12.99,3.0,2022-01,2022,52,5,Saturday,2021-W52
1,2,2022-01-01,28,1,1,Action Figure,Toys,9.99,15.99,Maven Toys Puebla 2,...,Downtown,2011-04-01,15.99,6.0,2022-01,2022,52,5,Saturday,2021-W52
2,3,2022-01-01,6,8,1,Deck Of Cards,Games,3.99,6.99,Maven Toys Mexicali 1,...,Commercial,2003-12-13,6.99,3.0,2022-01,2022,52,5,Saturday,2021-W52
3,4,2022-01-01,48,7,1,Dart Gun,Sports & Outdoors,11.99,15.99,Maven Toys Saltillo 2,...,Commercial,2016-03-23,15.99,4.0,2022-01,2022,52,5,Saturday,2021-W52
4,5,2022-01-01,44,18,1,Lego Bricks,Toys,34.99,39.99,Maven Toys Puebla 3,...,Residential,2014-12-27,39.99,5.0,2022-01,2022,52,5,Saturday,2021-W52


In [119]:
weekly_trend = toy_sales.groupby(['Year_Week' ,'Year',  'Day_Name'])[['Revenue' , 'Profit']].sum()

In [120]:
idx = weekly_trend.groupby('Year_Week')['Profit'].idxmax()

In [121]:
best_day = weekly_trend.loc[idx]

In [122]:
best_days_sort = best_day.sort_values(['Year' , 'Profit'] , ascending = (True , False))

In [118]:
best_days_sort.to_csv(r"D:\My Statistics\Pandas\Mexico toy sales result\best_days.csv")

In [123]:
best_days_sort

,,,Revenue,Profit
Year_Week,Year,Day_Name,,
2022-W51,2022,Saturday,50900.14,24800.64
2022-W17,2022,Saturday,47547.93,22316.56
2022-W24,2022,Saturday,38576.92,20902.54
2022-W37,2022,Friday,41578.46,18587.86
2022-W10,2022,Saturday,37479.05,17323.70
...,...,...,...,...
2023-W33,2023,Saturday,29331.75,12567.43
2023-W36,2023,Friday,32290.09,12314.81
2023-W38,2023,Friday,30007.98,11954.28


In [124]:
toy_sales.head()

,Sale_ID,Date,Store_ID,Product_ID,Units,Product_Name,Product_Category,Product_Cost,Product_Price,Store_Name,...,Store_Location,Store_Open_Date,Revenue,Profit,Month,Year,Week,Day,Day_Name,Year_Week
0,1,2022-01-01,24,4,1,Chutes & Ladders,Games,9.99,12.99,Maven Toys Aguascalientes 1,...,Downtown,2010-07-31,12.99,3.0,2022-01,2022,52,5,Saturday,2021-W52
1,2,2022-01-01,28,1,1,Action Figure,Toys,9.99,15.99,Maven Toys Puebla 2,...,Downtown,2011-04-01,15.99,6.0,2022-01,2022,52,5,Saturday,2021-W52
2,3,2022-01-01,6,8,1,Deck Of Cards,Games,3.99,6.99,Maven Toys Mexicali 1,...,Commercial,2003-12-13,6.99,3.0,2022-01,2022,52,5,Saturday,2021-W52
3,4,2022-01-01,48,7,1,Dart Gun,Sports & Outdoors,11.99,15.99,Maven Toys Saltillo 2,...,Commercial,2016-03-23,15.99,4.0,2022-01,2022,52,5,Saturday,2021-W52
4,5,2022-01-01,44,18,1,Lego Bricks,Toys,34.99,39.99,Maven Toys Puebla 3,...,Residential,2014-12-27,39.99,5.0,2022-01,2022,52,5,Saturday,2021-W52


In [125]:
store_revenue = pd.pivot_table(toy_sales , index = 'Store_Name' , columns = 'Product_Category' ,values = 'Revenue' , aggfunc = 'sum')

In [126]:
store_revenue['Total_Revenue'] = store_revenue.sum(axis = 1)

In [127]:
store_revenue.sort_values('Total_Revenue' , ascending = False)

Product_Category,Art & Crafts,Electronics,Games,Sports & Outdoors,Toys,Total_Revenue
Store_Name,,,,,,
Maven Toys Ciudad de Mexico 2,90076.29,112851.72,103915.27,81530.14,166180.01,554553.43
Maven Toys Guadalajara 3,81488.50,80031.12,67524.62,59489.39,160821.28,449354.91
Maven Toys Ciudad de Mexico 1,77326.87,42529.64,72040.97,72838.85,168819.88,433556.21
Maven Toys Toluca 1,108295.02,31583.30,52794.89,70867.01,147617.10,411157.32
Maven Toys Monterrey 2,64691.13,67519.80,61643.28,45251.07,133893.54,372998.82
Maven Toys Guadalajara 4,41273.42,77909.04,43952.90,37165.08,148166.20,348466.64
Maven Toys Hermosillo 3,69883.14,84740.47,49067.08,27958.86,113197.09,344846.64
Maven Toys Xalapa 2,82595.12,46296.85,45230.89,49633.21,120550.97,344307.04
Maven Toys Ciudad de Mexico 3,67190.32,30499.87,47150.87,72224.88,120358.72,337424.66


In [135]:
store_revenue.to_csv(r'D:\My Statistics\Pandas\Mexico toy sales result\store_revenue_pd_cat.csv')

In [128]:
product_analysis = pd.pivot_table(toy_sales , index = 'Month' , columns = 'Product_Name' , values = 'Revenue' , aggfunc = 'sum')

In [129]:
product_analysis.info()

<class 'pandas.core.frame.DataFrame'>
PeriodIndex: 21 entries, 2022-01 to 2023-09
Freq: M
Data columns (total 35 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Action Figure          21 non-null     float64
 1   Animal Figures         21 non-null     float64
 2   Barrel O' Slime        17 non-null     float64
 3   Chutes & Ladders       21 non-null     float64
 4   Classic Dominoes       21 non-null     float64
 5   Colorbuds              21 non-null     float64
 6   Dart Gun               21 non-null     float64
 7   Deck Of Cards          21 non-null     float64
 8   Dino Egg               18 non-null     float64
 9   Dinosaur Figures       21 non-null     float64
 10  Etch A Sketch          11 non-null     float64
 11  Foam Disk Launcher     7 non-null      float64
 12  Gamer Headphones       15 non-null     float64
 13  Glass Marbles          21 non-null     float64
 14  Hot Wheels 5-Pack      19 non-null     f

In [130]:
toy_sales.head()

,Sale_ID,Date,Store_ID,Product_ID,Units,Product_Name,Product_Category,Product_Cost,Product_Price,Store_Name,...,Store_Location,Store_Open_Date,Revenue,Profit,Month,Year,Week,Day,Day_Name,Year_Week
0,1,2022-01-01,24,4,1,Chutes & Ladders,Games,9.99,12.99,Maven Toys Aguascalientes 1,...,Downtown,2010-07-31,12.99,3.0,2022-01,2022,52,5,Saturday,2021-W52
1,2,2022-01-01,28,1,1,Action Figure,Toys,9.99,15.99,Maven Toys Puebla 2,...,Downtown,2011-04-01,15.99,6.0,2022-01,2022,52,5,Saturday,2021-W52
2,3,2022-01-01,6,8,1,Deck Of Cards,Games,3.99,6.99,Maven Toys Mexicali 1,...,Commercial,2003-12-13,6.99,3.0,2022-01,2022,52,5,Saturday,2021-W52
3,4,2022-01-01,48,7,1,Dart Gun,Sports & Outdoors,11.99,15.99,Maven Toys Saltillo 2,...,Commercial,2016-03-23,15.99,4.0,2022-01,2022,52,5,Saturday,2021-W52
4,5,2022-01-01,44,18,1,Lego Bricks,Toys,34.99,39.99,Maven Toys Puebla 3,...,Residential,2014-12-27,39.99,5.0,2022-01,2022,52,5,Saturday,2021-W52


In [131]:
### Store vs MOnth 

In [133]:
store_revenue_by_each_moth = pd.pivot_table(toy_sales , index = 'Store_Name' , columns = 'Month' , values = 'Revenue' , aggfunc = 'sum')

In [139]:
month_over_change = toy_sales.groupby('Month')['Revenue'].sum().pct_change()*100

In [140]:
month_over_change

Month
2022-01          NaN
2022-02    -0.221777
2022-03     8.891363
2022-04    15.536911
2022-05    -1.277848
2022-06    -1.545233
2022-07   -16.004404
2022-08   -11.979748
2022-09    19.701028
2022-10     6.491549
2022-11     5.999567
2022-12    32.647540
2023-01   -14.820671
2023-02    -3.287494
2023-03    22.263532
2023-04    -6.318459
2023-05    -0.286530
2023-06    -2.062261
2023-07     2.480469
2023-08   -20.217543
2023-09    -0.405914
Freq: M, Name: Revenue, dtype: float64

In [142]:
toy_sales.to_csv(r'D:\My Statistics\Pandas\Mexico toy sales result\toy_sales.csv')